python/pyspark/sql/connect/column.py

```diff
@@ -29,6 +29,17 @@
     from pyspark.sql.connect.client import SparkConnectClient
     import pyspark.sql.connect.proto as proto
 
+# TODO: solve the circular import between _typing and this class if we want to
```

Comment:
Should ideally have a JIRA for each todo.

BTW,  we can avoid the circular import by adding this import under `if TYPE_CHECKING`, and use the hint with strings. e.g.) `Union["PrimitiveType", "Column"])`